TrafficLight.py

```diff
@@ -0,0 +1,111 @@
+""" Represents the state of a traffic flow at a single light, a matrix made of rows that each
+represent a state of the traffic system. Within each row are floats between 0 and 1,
+representing the relative speed of traffic that can be processed in the column's direction to
+the direction of the row, IE a straight passage might be 1,
+a right turn might be 0.25, and a left turn might be 0.1""
+
+Also takes in a list of names exactly equal to the number of directions, with the nth name referring
+to activities within the nth row and column [name_list]
+Also takes in a list of timers with the length of each state listed in seconds (can be floats) [timers]
+"""
+
+
+class TrafficLight:
+    def __init__(self, name, traffic_matrix, traffic_flow_matrix_list, name_list, timers):
+        self.name = name
+        self.traffic_matrix = traffic_matrix
+        self.traffic_flow_matrix_list = traffic_flow_matrix_list
+        self.name_list = name_list
+        self.timers = timers
+        self.current_state = 0
+        self.elapsed_time_in_state= 0
+        if not self.checkInvariants(self):
+            raise TypeError("Invalid constructor for Traffic Light "+name)
+
+    """Ensures the state of the Traffic Flow Model follows all rules"""
+    def checkInvariants(self):
```

Comment:
Same as earlier can do this with assert statements, as they can be more informative to the user